server/data/data-jpa/src/main/java/org/apache/james/sieve/jpa/JPASieveRepository.java

```diff
@@ -0,0 +1,337 @@
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceException;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.backends.jpa.TransactionRunner;
+import org.apache.james.core.User;
+import org.apache.james.core.quota.QuotaSize;
+import org.apache.james.sieve.jpa.model.JPASieveQuota;
+import org.apache.james.sieve.jpa.model.JPASieveScript;
+import org.apache.james.sieverepository.api.ScriptContent;
+import org.apache.james.sieverepository.api.ScriptName;
+import org.apache.james.sieverepository.api.ScriptSummary;
+import org.apache.james.sieverepository.api.SieveRepository;
+import org.apache.james.sieverepository.api.exception.DuplicateException;
+import org.apache.james.sieverepository.api.exception.IsActiveException;
+import org.apache.james.sieverepository.api.exception.QuotaExceededException;
+import org.apache.james.sieverepository.api.exception.QuotaNotFoundException;
+import org.apache.james.sieverepository.api.exception.ScriptNotFoundException;
+import org.apache.james.sieverepository.api.exception.StorageException;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
+
+public class JPASieveRepository implements SieveRepository {
+    private static final String DEFAULT_SIEVE_QUOTA_USERNAME = "default.quota";
+
+    private final TransactionRunner transactionRunner;
+
+    @Inject
+    public JPASieveRepository(EntityManagerFactory entityManagerFactory) {
+        this.transactionRunner = new TransactionRunner(entityManagerFactory);
+    }
+
+    @Override
+    public void haveSpace(User user, ScriptName name, long size) throws QuotaExceededException, StorageException {
+        long usedSpace = findAllSieveScriptsForUser(user).stream()
+                .filter(sieveScript -> !sieveScript.getScriptName().equals(name.getValue()))
+                .mapToLong(JPASieveScript::getScriptSize)
+                .sum();
+
+        QuotaSize quota = limitToUse(user);
+        if (overQuotaAfterModification(usedSpace, size, quota)) {
+            throw new QuotaExceededException();
+        }
+    }
+
+    private QuotaSize limitToUse(User user) throws StorageException {
+        Optional<JPASieveQuota> userQuota = findQuotaForUser(user.asString());
+        if (userQuota.isPresent()) {
+            return QuotaSize.size(userQuota.map(JPASieveQuota::getSize));
+        }
+
+        Optional<JPASieveQuota> defaultQuota = findQuotaForUser(DEFAULT_SIEVE_QUOTA_USERNAME);
+        if (defaultQuota.isPresent()) {
+            return QuotaSize.size(defaultQuota.map(JPASieveQuota::getSize));
+        } else {
+            return QuotaSize.unlimited();
+        }
+    }
+
+    private boolean overQuotaAfterModification(long usedSpace, long size, QuotaSize quota) {
+        return QuotaSize.size(usedSpace).add(size).isGreaterThan(quota);
```

Comment:
Can you add one line break between each method calls, for example:
```
QuotaSize.size(usedSpace)
    .add(size)
    .isGreaterThan(quota);
```
this make the code a bit easier to be read.